src/recon_buildblock/PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin.cxx

```diff
@@ -122,13 +116,69 @@ set_num_subsets(const int new_num_subsets)
   return this->num_subsets;
 }
 
+template <typename TargetT>
+void
+PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin<TargetT>::
+set_proj_matrix(const shared_ptr<ProjMatrixByBin>& arg)
+{
+    this->PM_sptr = arg;
+}
+
+template<typename TargetT>
+void
+PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin<TargetT>::
+set_proj_data_info(const ProjData& arg)
+{
+    this->proj_data_info_sptr = arg.get_proj_data_info_sptr()->create_shared_clone();
+    if(this->skip_lm_input_file)
+    {
+        std::cout << "Dummy LM file" << std::endl;
+        this->list_mode_data_sptr.reset(new ListModeData_dummy(
+                                            arg.get_exam_info_sptr(),
+                                            proj_data_info_sptr));
+        this->frame_defs = arg.get_exam_info_sptr()->get_time_frame_definitions();
+    }
+    max_ring_difference_num_to_process = this->proj_data_info_sptr->get_scanner_sptr()->
+            get_num_rings() - 1;
+}
+
+template<typename TargetT>
+void
+PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin<TargetT>::
+set_skip_balanced_subsets(const bool arg)
+{
+  skip_balanced_subsets = arg;
+}
+
+template<typename TargetT>
+void
+PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin<TargetT>::
+set_max_ring_difference(const int arg)
+{
+    max_ring_difference_num_to_process = arg;
+    if (max_ring_difference_num_to_process > proj_data_info_sptr->get_max_segment_num())
+    {
+        error("In the parameter file, the 'maximum ring difference' is larger than the number of segments"
+                "in the listmode file. Abort.");
+    }
+    else if (max_ring_difference_num_to_process < proj_data_info_sptr->get_max_segment_num())
+    {
+        proj_data_info_sptr->reduce_segment_range(-max_ring_difference_num_to_process,
+                                                      max_ring_difference_num_to_process);
+    }
+}
+
+
 template<typename TargetT>
 bool
 PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin<TargetT>::
 actual_subsets_are_approximately_balanced(std::string& warning_message) const
 {
     if (skip_balanced_subsets)
+    {
+        warning(boost::format("We presume balaned subsets!"));
```

Comment:
```suggestion
        warning("We skip the check on balanced subsets and presume they are balanced!"));
```
although I don't see why we need this capability (can't take any appreciable time really?). fine to leave it for now

Reply:
I always find it very slow :D